### PR TITLE
productScreen and productCard

### PR DESCRIPTION
**Pull request by liran26**
finish productCard for now. added product screen & product card so i can see what i did.
beacuse i didn't getting yet a product id i used constant one.
there is no images in DB so i used two local one to test layout.

after i'll get those two from outside i'll test it again with the uncomments codes

### CODE REVIEW

File: components/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+const ProductCard = ({ navigation, productId }) => {
+    const [isFavorite, setIsFavorite] = useState(false); // State to track if the product is in favorites
+    const [product, setProduct] = useState(); // State to hold product data
+    const [sizePressed, setSizePressed] = useState(false); // State to track if size is pressed
+    const [colorPressed, setColorPressed] = useState(false); // State to track if color is pressed
+
+    useEffect(() => {
+        const fetchProduct = async () => {
+            try {
+                const productData = await getSpecificProduct("65faf26b8dec99b4cd38a768"); // This is the product ID for testing, real id will be passed from the products screen
```
Comment (GiliEllen) [inline components/ProductCard.js]: please fix, this is meant to be generic.

File: components/ProductCard.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    }, [navigation, product]);
+
+    // Rendering the ProductCard UI
+    return (
+        <View style={styles.container}>
+            <ScrollView vertical={false} horizontal={true} style={styles.imageContainer} showsHorizontalScrollIndicator={false} showsVerticalScrollIndicator={false}  >
+                {/* no images in DB so replaced with local for testing */}
+                <Image source={require('../assets/shirt1.png')} style={styles.image} /> 
+                <Image source={require('../assets/shirt2.png')} style={styles.image} />
+                {/* {product && product.images.map((image, index) => (
```
Comment (GiliEllen) [inline components/ProductCard.js]: remove comments

File: components/ProductCard.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                </Pressable>
+            ),
+        });
+    }, [navigation, product]);
+
+    // Rendering the ProductCard UI
+    return (
+        <View style={styles.container}>
+            <ScrollView vertical={false} horizontal={true} style={styles.imageContainer} showsHorizontalScrollIndicator={false} showsVerticalScrollIndicator={false}  >
+                {/* no images in DB so replaced with local for testing */}
```
Comment (GiliEllen) [inline components/ProductCard.js]: remove comments

File: components/ProductCard.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                        </View>
+                        <View>
+                            <Text style={{ fontSize: 24 }}>${product.price}</Text>
+                        </View>
+                    </View>
+                )}
+                {product && (
+                    <View style={styles.content}>
+                        <Text style={{ color: "#222222", fontSize: 14 }}>
+                            lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```
Comment (GiliEllen) [inline components/ProductCard.js]: what is this? remove

**Review by GiliEllen — CHANGES_REQUESTED** [review]
please follo the comments and fixe before updating PR

**Comment by liran26** [comment]
fixed all

File: App.js [hunk 5]
```diff
@@ -32,6 +35,8 @@ export default function App() {
         })}
       >
         <Tab.Screen name="Home" component={MainScreen} />
+        <Tab.Screen name="NotHome" component={MainScreen} />
+        <Tab.Screen name="productCard" component={ProductScreen} />
```
Comment (GiliEllen) [inline App.js]: remove

File: api/productsApi.js [hunk 6]
```diff
@@ -12,3 +12,15 @@ export const getAllProducts = async () => {
     return error;
   }
 };
+
+export const getSpecificProduct = async (productId) => {
+  try {
+    const { data } = await axios.get(`${API_URL}/api/products/${productId}`);
+    console.log(data);
```
Comment (GiliEllen) [inline api/productsApi.js]: remove

File: components/ProductCard.js [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    const [product, setProduct] = useState(); 
+    const [sizePressed, setSizePressed] = useState(false); 
+    const [colorPressed, setColorPressed] = useState(false); 
+
+    useEffect(() => {
+        const fetchProduct = async () => {
+            try {
+                const productData = await getSpecificProduct(productId); 
+                if (!productData) {
+                    console.error("Product not found");
```
Comment (GiliEllen) [inline components/ProductCard.js]: tell user, no item was found

File: components/ProductCard.js [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+                            </View>
+                        </View>
+                        <View>
+                            <Text style={{ fontSize: 24 }}>${product.price}</Text>
+                        </View>
+                    </View>
+                )}
+                {product && (
+                    <View style={styles.content}>
+                        <Text style={{ color: "#222222", fontSize: 14 }}>
```
Comment (GiliEllen) [inline components/ProductCard.js]: create object for styling

File: components/ProductCard.js [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+                            {product.description}
+                        </Text>
+                    </View>
+                )}
+            </View>
+            {/* Add to Cart button */}
+            {product && (
+                <Pressable style={styles.addToCart}>
+                    <Text style={{ color: "white", textAlign: "center" }}>ADD TO CART</Text>
```
Comment (GiliEllen) [inline components/ProductCard.js]: create object for styling

File: components/ProductCard.js [hunk 10]
```diff
@@ -0,0 +1,189 @@
+import { View, Text, Pressable } from 'react-native';
+import React, { useState, useEffect, useLayoutEffect } from 'react';
+import { Image, ScrollView, StyleSheet } from 'react-native';
+import { Ionicons } from '@expo/vector-icons'; 
+import { getSpecificProduct } from '../api/productsApi'; 
+
+const ProductCard = ({ navigation, productId }) => {
```
Comment (GiliEllen) [inline components/ProductCard.js]: remove navigation prop

File: components/ProductCard.js [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+    const [isFavorite, setIsFavorite] = useState(false); 
+    const [product, setProduct] = useState(); 
+    const [sizePressed, setSizePressed] = useState(false); 
+    const [colorPressed, setColorPressed] = useState(false); 
+
```
Comment (GiliEllen) [inline components/ProductCard.js:14]: const navigation = useNavigation();

**Review by GiliEllen — CHANGES_REQUESTED** [review]

**Review by GiliEllen — APPROVED** [review]

**Pull request merged**